### PR TITLE
adds docs for form layouts, finalizes layouts api

### PR DESCRIPTION
**Pull request by iway1**
@scamden ended up changing the api a little (removed the need for `renderedFields` unpacking, figured it'd save a bit of typing)

### CODE REVIEW

File: src/createSchemaForm.tsx [hunk 1]
```diff
@@ -462,9 +460,7 @@ export function createTsForm<
         <ActualFormComponent {...formProps} onSubmit={submitFn}>
           {renderBefore && renderBefore({ submit: submitFn })}
           {CustomChildrenComponent ? (
-            <CustomChildrenComponent
-              renderedFields={renderedFields}
-            ></CustomChildrenComponent>
+            <CustomChildrenComponent {...renderedFields} />
```
Comment (scamden) [inline src/createSchemaForm.tsx:463]: If we want to pass more props to this component later does it seem ok for them to be mixed in with the fields? Only concern I see is name collision, probably not a big worry though

File: www/docs/docs/custom-layouts.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+---
+
+# Custom Layouts
+
+## Defining Custom Layouts
+
+`react-ts-form` supports custom layouts for times when we need more than props just props to achieve the desired style (for example when we want two inputs in the same row).
```
Comment (scamden) [inline www/docs/docs/custom-layouts.md:8]: "Props just props"

File: www/docs/docs/custom-layouts.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+To build a form with a custom layout, pass a function that returns JSX as the child to your form component, and that function will be passed the form elements as a prop:
```
Comment (scamden) [inline www/docs/docs/custom-layouts.md:10]: Docs look great! Thank you for taking this!

File: src/createSchemaForm.tsx (hunk 1 above)
Comment (iway1) [reply src/createSchemaForm.tsx:463]: Yeah that's a good point but I can't think of anything else we'd want to pass here at the moment (not saying there isn't anything though) @scamden

if we end up needing it we can change it back it 2.0 😄

I really like having the API as simple as possible though, and I think not having to unpack is a little more dev friendly

Comment (scamden) [reply src/createSchemaForm.tsx:463]: Agreed :)

File: www/docs/docs/custom-layouts.md (hunk 2 above)
Comment (iway1) [reply www/docs/docs/custom-layouts.md:8]: Oof ty 🙃

File: www/docs/docs/custom-layouts.md (hunk 3 above)
Comment (iway1) [reply www/docs/docs/custom-layouts.md:10]: Thanks for your awesome solution! 🙏

**Pull request merged**